### PR TITLE
Add flag for trace sampling rate

### PR DESCRIPTION
**Pull request by RebeccaMahany**
Adds a new flag value type for float64s to support new flag `trace_sampling_rate`, which controls the fraction of traces that are sampled and exported.

Also starts backfilling some tests.

### CODE REVIEW

File: cmd/launcher/options.go [hunk 1]
```diff
@@ -83,6 +83,7 @@ func parseOptions(subcommandName string, args []string) (*launcher.Options, erro
 		flCompactDbMaxTx         = flagset.Int64("compactdb-max-tx", 65536, "Maximum transaction size used when compacting the internal DB")
 		flConfigFilePath         = flagset.String("config", defaultConfigFilePath, "config file to parse options from (optional)")
 		flExportTraces           = flagset.Bool("export_traces", false, "Whether to export traces")
+		flTraceSamplingRate      = flagset.Float64("trace_sampling_rate", 0.0, "What fraction of traces should be sampled")
```
Comment (directionless) [inline cmd/launcher/options.go:86]: My sense is that doing this as an integer is going to be less code, and that there's very little gain from a float. OTOH, you wrote the code, so 🤷

**Review by directionless — APPROVED** [review]
Generally feels fine. Not sure we really need the float

File: cmd/launcher/options.go (hunk 1 above)
Comment (RebeccaMahany) [reply cmd/launcher/options.go:86]: Unless I missed it, I don't think we support int flags yet in the `flags` package either, so it'd be the same amount of work: https://github.com/kolide/launcher/tree/main/pkg/agent/flags

Comment (directionless) [reply cmd/launcher/options.go:86]: Ah, indeed.

**Review by directionless — APPROVED** [review]

File: cmd/launcher/options.go (hunk 1 above)
Comment (directionless) [reply cmd/launcher/options.go:86]: I just like ints more than floats. 😆

**Pull request merged**